bundles/org.openhab.binding.freeathomesystem/src/main/java/org/openhab/binding/freeathomesystem/internal/handler/FreeAtHomeBridgeHandler.java

```diff
@@ -0,0 +1,819 @@
+                // Add authentication credentials.
+                AuthenticationStore auth = httpClient.getAuthenticationStore();
+
+                URI uri1 = new URI("http://" + ipAddress + "/fhapi/v1");
+                auth.addAuthenticationResult(new BasicAuthentication.BasicResult(uri1, username, password));
+
+                String url = "http://" + ipAddress + "/fhapi/v1/api/rest/devicelist";
+
+                Request req = httpClient.newRequest(url);
+                ContentResponse res = req.send();
+
+                // check status
+                if (res.getStatus() == 200) {
+                    // response OK
+                    httpConnectionOK.set(true);
+
+                    ret = true;
+
+                    logger.info("HTTP connection to SysAP is OK");
+                } else {
+                    // response NOK, set error
+                    httpConnectionOK.set(false);
+
+                    ret = false;
+
+                    logger.info("Wrong credentials for SysAP");
+                }
+            } catch (URISyntaxException | InterruptedException | ExecutionException | TimeoutException ex) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                        "Cannot open http connection, wrong passord");
+
+                logger.error("Cannot open http connection {}", ex.getMessage());
+
+                ret = false;
+            }
+        }
+
+        return ret;
+    }
+
+    /**
+     * Method to close Http connection
+     *
+     * @author Andras Uhrin
+     *
+     */
+    public boolean closeHttpConnection() {
+        boolean ret = false;
+
+        try {
+            // Start HttpClient.
+            switch (httpClient.getState()) {
+                case AbstractLifeCycle.FAILED:
+                case AbstractLifeCycle.STOPPING:
+                case AbstractLifeCycle.STOPPED: {
+                    break;
+                }
+                case AbstractLifeCycle.STARTING:
+                case AbstractLifeCycle.STARTED: {
+                    httpClient.stop();
+                    break;
+                }
+            }
+
+            logger.debug("Stop http client");
+
+            ret = true;
+        } catch (Exception ex) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, "Cannot stop http client");
+
+            logger.debug("Cannot stop http client - {}", ex.getMessage());
+
+            ret = false;
+        }
+
+        return ret;
+    }
+
+    /**
+     * Method to restart Http connection
+     *
+     * @author Andras Uhrin
+     *
+     */
+    public boolean restartHttpConnection() {
+        boolean ret = false;
+
+        ret = closeHttpConnection();
+
+        if (ret) {
+            ret = openHttpConnection();
+        }
+
+        return ret;
+    }
+
+    /**
+     * Method to connect the websocket session
+     *
+     * @author Andras Uhrin
+     *
+     */
+    @SuppressWarnings("null")
```

Comment:
It would be better to fix annotation issues rather than suppressing them.

Reply:
I will check if solution there I will change